source/Engine.cpp

```diff
@@ -1596,6 +1595,20 @@ void Engine::CalculateStep()
 	for(const Visual &visual : visuals)
 		batchDraw[calcTickTock].AddVisual(visual);
 
+	// Update fleets, remove fleets with no ships.
+	auto it = fleets.begin();
+	while(it != fleets.end())
+	{
+		bool shouldBeRemoved = true;
+		for(shared_ptr<Ship> ship : fleet.ships)
+			if(ship.get())
+				shouldBeRemoved = false;
+		if(shouldBeRemoved)
+			it = fleets.erase(it);
+		else
+			++it;
+	}
```

Comment:
This compiles on ubuntu 18 with g++

```suggestion
	for(auto it = fleets.begin(); it != fleets.end(); )
	{
		bool shouldBeRemoved = true;
		for(shared_ptr<Ship> ship : it->ships)
			if(ship.get())
			{
				shouldBeRemoved = false;
				break;
			}
		if(shouldBeRemoved)
			it = fleets.erase(it);
		else
			++it;
	}
```